### ISSUE TITLE
IDE not opening after working initially

### ISSUE THREAD

**Issue by mpgoebel** [hassio-addons/addon-ide#1]
# Problem/Motivation

IDE opens normally once or twice but after that it seems to not get the connection opened.

## Expected behavior

Open Web UI/ Click ipanel for IDE and it loads for use

## Actual behavior

Continually loads into perpetuity after closing previously

## Steps to reproduce

Open IDE then close IDE
Try opening again and it just sits on the loading screen

Below is and excerpt of the log from the add-on.  I have changed the IP addresses to mask personal info.

```
2017/11/28 10:11:25 [warn] 657#657: no resolver defined to resolve ocsp.int-x3.letsencrypt.org while requesting certificate status, responder: ocsp.int-x3.letsencrypt.org, certificate: "/ssl/fullchain.pem"
[28/Nov/2017:10:11:25 -0600] "GET /vfs/1/9ceCrjwKCINWb376/socket/?access_token=token&EIO=3&transport=polling&t=M03PRUu HTTP/1.1" 499 910 "https://My-ouside-IP.duckdns.org/ide" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Safari/537.36"
xx.xxx.xxx.xxx - - [28/Nov/2017:10:11:33 -0600] "GET / HTTP/1.1" 302 5 "https://My-ouside-IP.duckdns.org/ide" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Safari/537.36"
xx.xxx.xxx.xxx - - [28/Nov/2017:10:11:35 -0600] "GET /vfs/1/9ceCrjwKCINWb376/socket/?access_token=token&EIO=3&transport=polling&t=M03PTs7 HTTP/1.1" 499 910 "https://My-ouside-IP.duckdns.org/ide" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Safari/537.36"
xx.xxx.xxx.xxx - - [28/Nov/2017:10:11:36 -0600] "GET /vfs/1/9ceCrjwKCINWb376/socket/?access_token=token&EIO=3&transport=polling&t=M03PTyo HTTP/1.1" 499 910 "https://My-ouside-IP.duckdns.org/ide" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Safari/537.36"
xx.xxx.xxx.xxx - - [28/Nov/2017:10:11:36 -0600] "GET /vfs/1/9ceCrjwKCINWb376/socket/?access_token=token&EIO=3&transport=polling&t=M03PU1b HTTP/1.1" 499 910 "https://My-ouside-IP.duckdns.org/ide" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Safari/537.36"
xx.xxx.xxx.xxx - - [28/Nov/2017:10:11:36 -0600] "GET /vfs/1/9ceCrjwKCINWb376/socket/?access_token=token&EIO=3&transport=polling&t=M03PU6K HTTP/1.1" 499 910 "https://My-ouside-IP.duckdns.org/ide" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Safari/537.36"
xx.xxx.xxx.xxx - - [28/Nov/2017:10:11:36 -0600] "GET /vfs/1/9ceCrjwKCINWb376/socket/?access_token=token&EIO=3&transport=polling&t=M03PUAw HTTP/1.1" 499 910 "https://My-ouside-IP.duckdns.org/ide" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Safari/537.36"
xx.xxx.xxx.xxx - - [28/Nov/2017:10:11:37 -0600] "GET /vfs/1/9ceCrjwKCINWb376/socket/?access_token=token&EIO=3&transport=polling&t=M03PUFP HTTP/1.1" 499 910 "https://My-ouside-IP.duckdns.org/ide" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Safari/537.36"
xx.xxx.xxx.xxx - - [28/Nov/2017:10:11:37 -0600] "GET /vfs/1/9ceCrjwKCINWb376/socket/?access_token=token&EIO=3&transport=polling&t=M03PUJy HTTP/1.1" 499 910 "https://My-ouside-IP.duckdns.org/ide" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Safari/537.36"
xx.xxx.xxx.xxx - - [28/Nov/2017:10:11:37 -0600] "GET /vfs/1/9ceCrjwKCINWb376/socket/?access_token=token&EIO=3&transport=polling&t=M03PUOR HTTP/1.1" 499 910 "https://My-ouside-IP.duckdns.org/ide" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Safari/537.36"
xx.xxx.xxx.xxx - - [28/Nov/2017:10:11:38 -0600] "GET /vfs/1/9ceCrjwKCINWb376/socket/?access_token=token&EIO=3&transport=polling&t=M03PUSs HTTP/1.1" 499 910 "https://My-ouside-IP.duckdns.org/ide" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Safari/537.36"
xx.xxx.xxx.xxx - - [28/Nov/2017:10:11:38 -0600] "GET /vfs/1/9ceCrjwKCINWb376/socket/?access_token=token&EIO=3&transport=polling&t=M03PUXK HTTP/1.1" 499 910 "https://My-ouside-IP.duckdns.org/ide" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Safari/537.36"
xx.xxx.xxx.xxx - - [28/Nov/2017:10:11:38 -0600] "GET /vfs/1/9ceCrjwKCINWb376/socket/?access_token=token&EIO=3&transport=polling&t=M03PUbr HTTP/1.1" 499 910 "https://My-ouside-IP.duckdns.org/ide" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Safari/537.36"
xx.xxx.xxx.xxx - - [28/Nov/2017:10:11:39 -0600] "GET /vfs/1/9ceCrjwKCINWb376/socket/?access_token=token&EIO=3&transport=polling&t=M03PUkB HTTP/1.1" 499 910 "https://My-ouside-IP.duckdns.org/ide" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Safari/537.36"
xx.xxx.xxx.xxx - - [28/Nov/2017:10:11:40 -0600] "GET /vfs/1/9ceCrjwKCINWb376/socket/?access_token=token&EIO=3&transport=polling&t=M03PU-P HTTP/1.1" 499 910 "https://My-ouside-IP.duckdns.org/ide" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Safari/537.36"
xx.xxx.xxx.xxx - - [28/Nov/2017:10:11:42 -0600] "GET /vfs/1/9ceCrjwKCINWb376/socket/?access_token=token&EIO=3&transport=polling&t=M03PVUH HTTP/1.1" 499 910 "https://My-ouside-IP.duckdns.org/ide" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Safari/537.36"
2017/11/28 10:11:46 [info] 902#902: *759 client closed connection while waiting for request, client: xx.xxx.xxx.xxx server: 0.0.0.0:8321
```

**Comment by mpgoebel** [comment]
I am also getting a frequent log event like 
`PTY is not supported.`

**Comment by mpgoebel** [comment]
When using IDE in an ipanel frame it happens only when you have opened IDE in a a new tab, changed to a different panel, then try to change back to IDE and it will have this bug, if I open home assistant in a new tab and try, it will open correctly again.

**Comment by frenck** [comment]
Fixed! Just released v0.3.0 which addresses this issue (hopefully).

If the error still occurs, feel free to re-open.

Thank you for the report.
